### PR TITLE
Circuit Improvements

### PR DESCRIPTION
**Pull request by saleel**
## Motivation

Use different technique to extract data for computing nullifiers with fewer constraints

## Change Summary

- Optimise data extraction circuits (reduces about 60% of the constraints)
- Add packing to make it easier to reveal data
- Update circuits to match V2 spec
- Changes to identityNullifier and photoNullifier definitions

## Merge Checklist

_Choose all relevant options below by adding an `x` now or at any time before submitting for review_

- [ ] PR title adheres to the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) standard
- [ ] PR has a [change set](https://github.com/privacy-scaling-explorations/anon-aadhaar/CHANGELOG.md)
- [ ] PR has been tagged with a change label(s) (i.e. documentation, feature, bug fix, or chore)
- [ ] PR includes documentation if necessary.

## Additional Context

If this is a relatively large or complex change, provide more details here that will help reviewers

### CODE REVIEW

File: packages/circuits/src/aadhaar-verifier.circom [hunk 1]
```diff
@@ -1,111 +1,73 @@
...
-    signal input aadhaarDataLength;             // length of the padded data
-    signal input signature[k];                  // RSA signature
-    signal input pubKey[k];                     // RSA public key (of the government)
+    signal input qrDataPadded[maxDataLength];
+    signal input qrDataPaddedLength;
+    signal input nonPaddedDataLength;
+    signal input delimiterIndices[18];
+    signal input signature[k];
+    signal input pubKey[k];
+    signal input appId;
```
Comment (Meyanis95) [inline packages/circuits/src/aadhaar-verifier.circom:33]: Don't you want to change to `nullifierSeed`?

**Review by Meyanis95 — APPROVED** [review]
LGTM! 
I can take it form here
The main concern I have is, do we merge it directly to `main` or we work on `develop`?

File: packages/circuits/src/aadhaar-verifier.circom (hunk 1 above)
Comment (saleel) [reply packages/circuits/src/aadhaar-verifier.circom:33]: Lets do that in another PR. Its not just about the rename I think - we will need to make this input private and then add another output which is the hash(nullifierSeed).
So, lets review the idea again and do in another PR.

Comment (Meyanis95) [reply packages/circuits/src/aadhaar-verifier.circom:33]: All right!

**Comment by aguzmant103** [comment]
@gitpoap-bot @saleel

**Pull request merged**